File: Application/Services/Operations/Finances/Dtos/CreditCardExpenses/CreditCardExpenseDto.cs
using System;
using Application.Services.Operations.Finances.Dtos.Bank;
using Application.Services.Operations.Finances.Dtos.CategorySubcategoryExpenses;
using Application.Services.Shared.Dtos;

namespace Application.Services.Operations.Finances.Dtos.CreditCardExpenses
{
    public class CreditCardExpenseDto:RootBaseDto
    {
        public string Name { get; set; }
        public int CategoryExpenseId { get; set; }
        public CategoryExpenseDto CategoryExpense { get; set; }
        public int SubcategoryExpenseId { get; set; }
        public SubcategoryExpenseDto SubcategoryExpense { get; set; }
        public int? PaidFromBankAccountId { get; set; }
        public BankAccountDto PaidFromBankAccount { get; set; }
        public int CardId { get; set; }
        public CardDto Card { get; set; }
        public decimal Price { get; set; }
        public DateTime Expires { get; set; }
        public DateTime WasPaid { get; set; }
        public string OthersPaymentMethods { get; set; }
        public string Document { get; set; }
        public string Description { get; set; }
        public int InstallmentsQuantity { get; set; }
        public decimal InstallmentPrice { get; set; }
        public decimal TotalPriceInterest { get; set; }
        public decimal TotalPercentageInterest { get; set; }
        public decimal PaymentAtSight { get; set; }
        public string CurrentInstallment { get; set; }
        public DateTime ExpenseDay { get; set; }
        public CreditCardLimitOperationDto CreditCardLimitOperation { get; set; }
        public int? CreditCardExpenseInvoiceId { get; set; }
        public CreditCardExpenseInvoiceDto CreditCardExpenseInvoice { get; set; }
    }
}

File: Views/src/components/financial/components/credit-card-fixed-expenses/dto/credit-card-expense-dto.ts
import { RootBase } from "src/shared/entities-dtos/root-base";
import { CardDto } from "../../bank-account-cards/dto/card-dto";
import { CreditCardLimitOperationDto } from "../../bank-account-cards/dto/credit-card-limit-operation-dto";
import { CategoryExpenseDto } from "../../common-components/category-subcategory-expenses/dto/category-expense-dto";
import { SubcategoryExpenseDto } from "../../common-components/category-subcategory-expenses/dto/subcategory-expense-dto";
import { CreditCardExpenseInvoiceDto } from "./credit-card-expense-invoice-dto";

export class CreditCardExpenseDto extends RootBase {
    name: string;
    categoryExpenseId:number;
    categoryExpense: CategoryExpenseDto;
    subcategoryExpenseId:number;
    subcategoryExpense:SubcategoryExpenseDto;
    cardId: number;
    card: CardDto;
    price: number;
    interest: number;
    expires: Date;
    wasPaid: Date;
    othersPaymentMethods: string;
    document: string;
    description: string;
    installmentsQuantity: number;
    totalPriceInterest: number;
    totalPercentageInterest: number;
    paymentAtSight: number;
    installmentPrice: number;
    currentInstallment: string;
    creditCardLimitOperation: CreditCardLimitOperationDto;
    creditCardExpenseInvoiceId: number;
    creditCardExpenseInvoice: CreditCardExpenseInvoiceDto;
    expenseDay: Date;
}
